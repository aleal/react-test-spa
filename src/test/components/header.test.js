import React from 'react';
import {renderComponent} from '../test-helper';
import Header from '../../components/header';

describe('App Index Tests' , () => {
    
    let component;
  
    it('renders header when user is not signed in', () => {
        component = renderComponent(Header);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders header when user is signed in', () => {
        component = renderComponent(Header,{}, {userData:{user:{name:'Test 123'}}});
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});