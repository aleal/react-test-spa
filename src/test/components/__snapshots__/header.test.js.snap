// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Index Tests renders header when user is not signed in 1`] = `
<div
  className="container"
>
  <div
    className="navbar navbar-expand-lg navbar-dark bg-dark fixed-top"
  >
    <a
      className="pull-xs-left "
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Logo"
        className="logo"
        src="/logo.png"
      />
    </a>
    <span
      className="pull-xs-right"
    >
      <a
        className="btn"
        href="/sign-in"
        onClick={[Function]}
      >
         Sign in 
      </a>
      <a
        className="btn"
        href="/sign-up"
        onClick={[Function]}
      >
         Sign up 
      </a>
    </span>
  </div>
  <hr />
</div>
`;

exports[`App Index Tests renders header when user is signed in 1`] = `
<div
  className="container"
>
  <div
    className="navbar navbar-expand-lg navbar-dark bg-dark fixed-top"
  >
    <a
      className="pull-xs-left "
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Logo"
        className="logo"
        src="/logo.png"
      />
    </a>
    <span
      className="pull-xs-right"
    >
      <span>
        Hello, 
        Test 123
         
      </span>
      <a
        className="btn"
        href="/sign-out"
        onClick={[Function]}
      >
         Sign out 
      </a>
    </span>
  </div>
  <hr />
</div>
`;
